clang-tools-extra/clang-tidy/modernize/UseDesignatedInitializersCheck.cpp

```diff
@@ -119,13 +119,18 @@ UseDesignatedInitializersCheck::UseDesignatedInitializersCheck(
 void UseDesignatedInitializersCheck::registerMatchers(MatchFinder *Finder) {
   const auto HasBaseWithFields =
       hasAnyBase(hasType(cxxRecordDecl(has(fieldDecl()))));
+
+  // see #133715
+  const auto IsSTLArray =
+      hasType(qualType(hasDeclaration(recordDecl(hasName("::std::array")))));
```

Comment:
I can imagine this issue can trigger on other similar cases, for example if one creates their own `std::array` class.

Can we make this a bit more generic, for example not warn of 1) the class is an aggregate and 2) it contains only one member and 3) that member is an array type?